operator/controllers/lifecycle/keptntaskdefinition/controller.go

```diff
@@ -58,13 +58,40 @@ func (r *KeptnTaskDefinitionReconciler) Reconcile(ctx context.Context, req ctrl.
 		r.Log.Error(err, "Failed to get the KeptnTaskDefinition")
 		return ctrl.Result{Requeue: true, RequeueAfter: 30 * time.Second}, nil
 	}
+	defSpec := common.GetRuntimeSpec(definition)
+	if definition.Spec.Container == nil && defSpec != nil { //if the spec is well-defined
 
-	if !reflect.DeepEqual(definition.Spec.Function, klcv1alpha3.FunctionSpec{}) {
-		err := r.reconcileFunction(ctx, req, definition)
+		// get configmap reference either existing configmap name or inline generated one
+		cmName := r.getCmName("keptnfn-"+definition.Name, defSpec)
+
+		//get existing configmap either generated from inline or user defined
+		cm, err := r.getConfigMap(ctx, cmName, req.Namespace)
+		//if IsNotFound we need to create it
+		if err != nil && !errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+
+		// generate the updated config map, this is either the existing config map or the inline one
+		functionCm := cm
+		if common.IsInline(defSpec) {
+			functionCm = r.generateConfigMap(defSpec, cmName, definition.Namespace)
+		}
+		//compare and handle updated and existing
+		r.reconcileConfigMap(ctx, functionCm, cm)
+		/// if neither exist remove from status
+		r.updateTaskDefinitionStatus(functionCm, definition)
+		// update ref
+		//defSpec.ConfigMapReference.Name = functionCm.Name
```

Comment:
since the `updateTaskDefinitionStatus()` function only updates the name of the referenced config map in the deprecated `Function` property of the spec, i think we need to uncomment this line again?

Reply:
was misreading this, so it seems this line can be removed